plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -739,7 +739,9 @@ class FossId internal constructor(
                     // stays in state "NOT FINISHED". Therefore, we check the output of the Git fetch to find out
                     // whether the download is actually done.
                     val message = response.message
-                    if (message == null || !GIT_FETCH_DONE_REGEX.containsMatchIn(message)) return@wait false
+                    if (version >= "20.2" || message == null || !GIT_FETCH_DONE_REGEX.containsMatchIn(message)) {
```

Comment:
I don't think the `>=` comparison on strings does what you want it to do. You probably need to use the `Semver` class for a proper version number comparison.

Reply:
Well we are already using this kind of comparison in other places: https://github.com/oss-review-toolkit/ort/blob/main/plugins/scanners/fossid/src/main/kotlin/FossId.kt#L703

Do you have a concrete example where the comparison would not be the correct one ?

Reply:
Ah I see this is the case "1" > "100" or something ?
Currently FossID versions have been pretty simple such as 20.2, 21.3.2, 23.3.0 so it should be ok.

As we urgently need this fix to repair our production environment, could we merge this and do the refactoring to`Semver" in a future PR ?

Reply:
> Do you have a concrete example where the comparison would not be the correct one ?

E.g. `"20.10" >= "20.2"` would return `false`, though it should be `true`.

Reply:
> As we urgently need this fix to repair our production environment

Please consider decoupling our production environment from upstream development to avoid such situations in the future.

> could we merge this and do the refactoring to`Semver" in a future PR ?

As you need to fix other places as well anyway, I'm ok with that. Please try to file that fixup PR before tomorrow's release.

Reply:
> You probably need to use the `Semver` class for a proper version number comparison.

Note to myself, another option could be to use Java's built-in [ModuleDescriptor.Version](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/module/ModuleDescriptor.Version.html).